drivers/storage/portworx/components_test.go

```diff
@@ -389,6 +401,37 @@ func TestBasicInstallWithPortworxDisabled(t *testing.T) {
 	require.Empty(t, dsList.Items)
 }
 
+func TestBasicInstallWithInternalKVDBDisabled(t *testing.T) {
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	reregisterComponents()
+	k8sClient := testutil.FakeK8sClient()
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+			Annotations: map[string]string{
+				pxutil.AnnotationPVCController: "false",
+			},
+		},
+		Spec: corev1.StorageClusterSpec{
+			Kvdb: &corev1.KvdbSpec{
+				Internal: false,
+			},
+		},
+	}
+
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Portworx Services should not be created
```

Comment:
Why it should not be created?

Reply:
.. because internal kvdb is disabled

We can improve the comment to something like - `Portworx kvdb service should not be created`